### PR TITLE
Extract constant URLs to own module

### PR DESCRIPTION
**Pull request by mfisher87**

### CODE REVIEW

File: icepyx/core/urls.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from typing import Final
+
+CMR_BASE_URL: Final = "https://cmr.earthdata.nasa.gov"
+GRANULE_SEARCH_BASE_URL: Final = f"{CMR_BASE_URL}/search/granules"
+COLLECTION_SEARCH_BASE_URL: Final = f"{CMR_BASE_URL}/search/collections.json"
+
+EGI_BASE_URL: Final = "https://n5eil02u.ecs.nsidc.org/egi"
+ORDER_BASE_URL: Final = f"{EGI_BASE_URL}/request"
+
+DOWNLOAD_BASE_URL: Final = "https://n5eil02u.ecs.nsidc.org/esir"
```
Comment (JessicaS11) [inline icepyx/core/urls.py:10]: Having just been working on incorporating GEDI into icepyx, I'm noting that these urls are NSIDC specific even though the names are more generic. It's not a dealbreaker, but something we'll have to confront in the near future. 

More generally, I didn't see any immediate analog in earthaccess, but moving towards directly using any urls that are easily surfaceable there is saving us work down the line.

**Review by JessicaS11 — APPROVED** [review]

**Comment by mfisher87** [comment]
Hey @JessicaS11 now that we've merged #547, can you give this another review to trigger integration tests?

**Review by JessicaS11 — APPROVED** [review]

**Comment by mfisher87** [comment]
Looks like something isn't right :) Looking in to it!

**Comment by JessicaS11** [comment]
> Looks like something isn't right :) Looking in to it!

I'm noticing that the checks ran (successfully, and interestingly included the integration test) after the precommit auto fixes (https://github.com/icesat2py/icepyx/pull/591/commits/06564c11b8083a7c782f215c1867373f023909e5). My approval triggered the UML auto-update (but not the integration test?), but [that commit](https://github.com/icesat2py/icepyx/pull/591/commits/cbaff78ec90c84df92208865711cc0ae2d8345ba) only triggered readthedocs and precommit checks. I'm guessing the status is missing because for some reason the tests weren't triggered for the head commit, but I don't quite understand why.

**Comment by mfisher87** [comment]
I especially don't understand why the unit tests weren't triggered!

**Review by JessicaS11 — APPROVED** [review]

**Comment by JessicaS11** [comment]
Maybe open PRs are just going to be a bit grumpy at us until we're into ones opened after the actions were implemented?

**Comment by mfisher87** [comment]
Weird. Maybe you're right and all will be fine from here!

I opened #595 in case this continues to give us trouble. I want to avoid spending too much of our limited time getting things perfect :)

**Pull request merged**